src/pybind/mgr/dashboard/frontend/src/app/shared/pipes/relative-date.pipe.ts

```diff
@@ -1,15 +1,30 @@
 import { Pipe, PipeTransform } from '@angular/core';
 
+import _ from 'lodash';
 import moment from 'moment';
 
 @Pipe({
-  name: 'relativeDate'
+  name: 'relativeDate',
+  pure: false
 })
 export class RelativeDatePipe implements PipeTransform {
-  transform(value: any): any {
-    if (!value) {
-      return 'unknown';
+  /**
+   * Convert a time into a human readable form, e.g. '2 minutes ago'.
+   * @param {Date | string | number} value The date to convert, should be
+   *   an ISO8601 string, an Unix timestamp (seconds) or Date object.
+   * @return {string} The time in human readable form or an empty string
+   *   on failure (e.g. invalid input).
+   */
+  transform(value: Date | string | number): string {
+    let date: moment.Moment;
```

Comment:
Maybe for a follow-up PR: as Moment is now considered a legacy project in maintenance mode, there is a tracker issue for its removal:
https://tracker.ceph.com/issues/47556

Reply:
Yes i know, but i think we should remove it in a separate PR. This will make it easier if we plan to backport this PR here.